### PR TITLE
Bugfix 500 on register and fix validation on firstname and lastname

### PR DESCRIPTION
**Pull request by maxXximus23**
https://trello.com/c/xjLH6G2S/21-be-fix-500-on-register-endpoint-when-validation-fails
https://trello.com/c/K6UZjSaR/22-be-fix-validation-for-firstname-and-lastname-on-register-user-endpoint

Changelog:
- Changed used ApplicationException on AppValidationException in AuthService;
- Added a new check on RegisterRequestValidator and PasswordRequestValidator for Firstname, Lastname, and Password if it contains whitespace.

### CODE REVIEW

File: src/WheelsAndGoods.Api/Program.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using FluentValidation;
```
Comment (illyxancuh) [inline src/WheelsAndGoods.Api/Program.cs]: Why we add FluentValidator in program.cs?

File: src/WheelsAndGoods.Api/Validation/Internal/PasswordRequestValidator.cs [hunk 2]
```diff
@@ -16,6 +16,8 @@ public PasswordRequestValidator()
 
 		RuleFor(model => model.Password)
 			.NotEmpty()
+			.Must(lastName => !lastName.Contains(' '))
+			.WithMessage("{PropertyName} contain a whitespace")
```
Comment (illyxancuh) [inline src/WheelsAndGoods.Api/Validation/Internal/PasswordRequestValidator.cs:20]: Typo
```
.Must(password=> !password.Contains(' '))
```

**Review by illyxancuh — CHANGES_REQUESTED** [review]

File: src/WheelsAndGoods.Api/Program.cs (hunk 1 above)
Comment (maxXximus23) [reply src/WheelsAndGoods.Api/Program.cs]: Good comment, will be removed.

File: src/WheelsAndGoods.Api/Validation/Internal/PasswordRequestValidator.cs (hunk 2 above)
Comment (maxXximus23) [reply src/WheelsAndGoods.Api/Validation/Internal/PasswordRequestValidator.cs:20]: The same thing, thanks.

**Review by kirpichyov — APPROVED** [review]

**Review by illyxancuh — APPROVED** [review]

**Pull request merged**